Integrated data storage solution

catilgan opened the pull request:
- Saving the pdf and extracted text in data storage
- Added a new env variable for data storage location (defaults to '<ROOT>/data')
- Returning language, pdf and text filenames as json object in POST request

Already deployed this solution to production for testing:
- https://extractor.bizres.ch/docs/index.html#/PDF/post_extractor

The pdf and text files are succesfully saved in our NFS storage folder.

dev-ng reviewed (approved):
Good job!

Two comments:
1. I think as soon as we allow write access we need to add more security = authorization.

2. use of swagger.yaml/swagger.json - Swagger UI needs .json file, but I find .yaml a bit more user friendly for editing. So I worked on .yaml and then converted it into  .json. We need to agree whether we need the .yaml file or stay with .json only to avoid confusions in the future.

catilgan commented:
> Good job!
> 
> Two comments:
> 
>     1. I think as soon as we allow write access we need to add more security = authorization.
> 
>     2. use of swagger.yaml/swagger.json - Swagger UI needs .json file, but I find .yaml a bit more user friendly for editing. So I worked on .yaml and then converted it into  .json. We need to agree whether we need the .yaml file or stay with .json only to avoid confusions in the future.

Hi Nick,

As discussed, I added a Basic-Authentication as the default authentication method. We can other authentication methods like Token-Auth later. In development environmen, you can find the basic authentication parameters in docker-compose.yaml.

For the production, I've added the environment variables in Hidora node environment settings. Furthermore, I've saved the credentials in a KeePassXC [1] database and synced it into Google Drive [2] for sharing with the dev team.

As personal preference, i like the YAML file structure, which is better readable and editable. I don't mind to converting it to JSON format manually.

Cheers, Cahit

[1] https://keepassxc.org/download/
[2] Google Drive > Product Development > *.kdbx Database

dev-ng commented:
I approve BasicAuth as a quick solution. I suggest to have a look at casbin as an alternative.

I see you updated the yaml file, but not the json :) this is where the devil hidden! The UI needs json...
